### PR TITLE
wire up telemetry for acuqire token silent

### PR DESCRIPTION
**Pull request by DarylThayil**
Add instrumentation for Acquire token silent.
This will be to help debug perf, but future instrumentation will be added

### CODE REVIEW

File: lib/msal-core/src/UserAgentApplication.ts [hunk 1]
```diff
@@ -40,6 +41,7 @@ import { Constants,
     PersistentCacheKeys,
     ErrorCacheKeys,
 } from "./utils/Constants";
+import uuid from "uuid";
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: Any reason we don't use the in house guid creator?

Comment (jasonnutter) [inline lib/msal-core/src/UserAgentApplication.ts]: Use `CryptoUtils` helper function to create a guid instead.

File: lib/msal-core/test/UserAgentApplication.spec.ts [hunk 2]
```diff
@@ -187,15 +187,15 @@ describe("UserAgentApplication.ts Class", function () {
...
-        });
+        // it("telemetry manager doesn't exis in UAA when not configured", () => {
+        //     msal = new UserAgentApplication({
+        //         auth: {
+        //             clientId: TEST_CONFIG.MSAL_CLIENT_ID
+        //         }
+        //     });
+        //     // @ts-ignore
+        //     expect(msal.telemetryManager).to.be.null;
+        // });
```
Comment (jasonnutter) [inline lib/msal-core/test/UserAgentApplication.spec.ts]: Remove instead.

File: samples/VanillaJSTestApp/JavascriptSPA/authConfig.js [hunk 3]
```diff
@@ -2,12 +2,22 @@
...
     cache: {
         cacheLocation: "localStorage", // This configures where your cache will be stored
         storeAuthStateInCookie: false, // Set this to "true" if you are having issues on IE11 or Edge
+    },
+    system: {
+        telemetry: {
+            applicationName: 'msalVanillaTestApp',
+            applicationVersion: 'test1.0',
+            telemetryEmitter: (events) => {
+                console.log(events);
```
Comment (jasonnutter) [inline samples/VanillaJSTestApp/JavascriptSPA/authConfig.js]: I would doing something like `console.log("Telemetry Event: ", event)` to differentiate from other logs.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 4]
```diff
@@ -589,7 +591,11 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(userRequest: AuthenticationParameters): Promise<AuthResponse> {
-
+        const requestCorrelationId = userRequest.correlationId || uuid.v4();
+        const apiEvent: ApiEvent = new ApiEvent(requestCorrelationId, this.logger);
+        apiEvent.apiEventIdentifier = API_EVENT_IDENTIFIER.AcquireTokenSilent;
+        apiEvent.apiCode = API_CODE.AcquireTokenSilent;
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: Is it possible to flatten this to a single constructor call? Having public members set without getters or setters is a bad code smell IMO.

File: lib/msal-core/src/UserAgentApplication.ts [hunk 5]
```diff
@@ -710,10 +716,21 @@ export class UserAgentApplication {
...
-        });
+        })
+            .then(res => {
+                apiEvent.wasSuccessful = true;
+                return res;
+            })
+            .catch((error: AuthError) => {
+                this.cacheStorage.resetTempCacheItems(request.state);
+                apiEvent.apiErrorCode = error.errorCode;
+                apiEvent.wasSuccessful = false;
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: make this a function on apiEvent instead

File: lib/msal-core/src/UserAgentApplication.ts [hunk 6]
```diff
@@ -710,10 +716,21 @@ export class UserAgentApplication {
                         return null;
                     });
             }
-        }).catch((error: AuthError) => {
-            this.cacheStorage.resetTempCacheItems(request.state);
-            throw error;
-        });
+        })
+            .then(res => {
+                apiEvent.wasSuccessful = true;
```
Comment (pkanher617) [inline lib/msal-core/src/UserAgentApplication.ts]: make this a function on apiEvent instead

**Review by jasonnutter — CHANGES_REQUESTED** [review]

File: lib/msal-core/src/UserAgentApplication.ts (hunk 4 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: I have thought about that, I am happy to revisit the design when we instrument more, but probably not in this pr

File: samples/VanillaJSTestApp/JavascriptSPA/authConfig.js (hunk 3 above)
Comment (DarylThayil) [reply samples/VanillaJSTestApp/JavascriptSPA/authConfig.js]: addiong

File: lib/msal-core/src/UserAgentApplication.ts (hunk 1 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: great call, fixing

Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: updated

File: lib/msal-core/src/UserAgentApplication.ts (hunk 5 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: same comment on changing design in another pr

File: lib/msal-core/src/UserAgentApplication.ts (hunk 6 above)
Comment (DarylThayil) [reply lib/msal-core/src/UserAgentApplication.ts]: same comment on changing design in another pr

File: lib/msal-core/test/UserAgentApplication.spec.ts (hunk 2 above)
Comment (DarylThayil) [reply lib/msal-core/test/UserAgentApplication.spec.ts]: Updated tests to replace these

**Review by pkanher617 — APPROVED** [review]
Might be helpful to see some docs for this as well. Approving for now.

File: lib/msal-core/src/UserAgentApplication.ts (hunk 4 above)
Comment (sameerag) [reply lib/msal-core/src/UserAgentApplication.ts]: On a side note, we also worked very hard to simplify this API :-), so it helps instrumenting the code in independent helpers.

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**